### PR TITLE
first step

### PR DESCRIPTION
**Pull request by Emptyyw**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import './App.css';
+import { Text } from '@mantine/core';
+import { ActionToggle } from './components/shared/ActionToggle/ActionToggle';
+import '@mantine/core/styles.css';
+
+function App() {
+  return (
+    <div>
```
Comment (AdammmVV) [inline src/App.tsx]: what is the div for?

File: src/App.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import './App.css';
+import { Text } from '@mantine/core';
+import { ActionToggle } from './components/shared/ActionToggle/ActionToggle';
+import '@mantine/core/styles.css';
+
+function App() {
```
Comment (AdammmVV) [inline src/App.tsx]: use the arrow function

File: src/components/shared/ActionToggle/ActionToggle.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    getInitialValueInEffect: true,
+  });
+  const toggleColorScheme = () => {
+    setColorScheme(
+      computedColorScheme === colorScheme.Light ? colorScheme.Dark : colorScheme.Light,
+    );
+  };
+
+  return (
+    <Group className={cx(classes.group)} justify="end">
```
Comment (AdammmVV) [inline src/components/shared/ActionToggle/ActionToggle.tsx]: what are `classes.group` for?

File: src/components/shared/ActionToggle/ActionToggle.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import cx from 'clsx';
+import {
+  ActionIcon,
+  useMantineColorScheme,
+  useComputedColorScheme,
+  Group,
+} from '@mantine/core';
+import { IconSun, IconMoon } from '@tabler/icons-react';
+import classes from './ActionToggle.module.css';
+import { colorScheme } from '../../../Enum/Enum';
```
Comment (AdammmVV) [inline src/components/shared/ActionToggle/ActionToggle.tsx]: use absolute imports

File: src/App.tsx [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+import '@mantine/core/styles.css';
+import './components//18next/i18n';
+import { useTranslation } from 'react-i18next';
+import LanguageSwitcher from 'components/shared/LanguageSwitcher/LanguageSwitcher';
+
+const App = () => {
+  const { t } = useTranslation();
+
+  return (
+    <>
```
Comment (AdammmVV) [inline src/App.tsx]: why ReactFragmet here?

File: src/components/18next/i18n.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    translation: {
+      Hello: 'Привет',
+    },
+  },
+} as const;
+
+i18n.use(initReactI18next).init({
+  resources,
+  lng: 'en',
+  fallbackLng: 'en',
```
Comment (AdammmVV) [inline src/components/18next/i18n.ts]: use Enum `langSwitcher` for `en`

File: src/components/18next/i18n.ts [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+  interpolation: {
+    escapeValue: false,
+  },
+});
+
+export default i18n;
+
+export type MyTranslationKeys = keyof (typeof resources)['ru']['translation'];
```
Comment (AdammmVV) [inline src/components/18next/i18n.ts]: use Enum `langSwitcher` for `ru`

File: src/components/shared/ActionToggle/ActionToggle.tsx [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+  useComputedColorScheme,
+  Group,
+} from '@mantine/core';
+import { IconSun, IconMoon } from '@tabler/icons-react';
+import classes from './ActionToggle.module.css';
+import { colorScheme } from 'Enum/Enum';
+
+export function ActionToggle() {
+  const { setColorScheme } = useMantineColorScheme();
+  const computedColorScheme = useComputedColorScheme('light', {
```
Comment (AdammmVV) [inline src/components/shared/ActionToggle/ActionToggle.tsx:14]: use Enum for `light`

File: src/components/shared/LanguageSwitcher/LanguageSwitcher.tsx [hunk 9]
```diff
@@ -0,0 +1,34 @@
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { ActionIcon, Group, Text } from '@mantine/core';
+import cx from 'clsx';
+import classes from './LanguageSwitcher.module.css';
+import { langSwitcher } from 'Enum/Enum';
+
+function LanguageSwitcher() {
+  const { i18n } = useTranslation();
```
Comment (AdammmVV) [inline src/components/shared/LanguageSwitcher/LanguageSwitcher.tsx:15]: save settings for theme and language in localStorage

File: src/App.tsx [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+import Dashboard from 'components/dashboard';
+import 'firebase/firebase';
+
+const App = () => {
+  return (
+    <Router>
+      <HeaderMenu />
+      <Container>
+        <Routes>
+          <Route path="/login" element={<AuthenticationForm type="login" />} />
```
Comment (AdammmVV) [inline src/App.tsx]: create enum for paths and for types `type="login"`

File: src/components/Auth/AuthForm.tsx [hunk 11]
```diff
@@ -0,0 +1,201 @@
...
+  });
+
+  const dispatch = useDispatch();
+
+  const handleRegister = event => {
+    event.preventDefault();
+    const auth = getAuth();
+    const email = form.values.email;
+    const password = form.values.password;
+    createUserWithEmailAndPassword(auth, email, password)
```
Comment (AdammmVV) [inline src/components/Auth/AuthForm.tsx]: put it in api file

File: src/components/Auth/AuthForm.tsx [hunk 12]
```diff
@@ -0,0 +1,201 @@
...
+          setUser({
+            login: form.values.login,
+            email,
+            token: userCredential.user.refreshToken,
+            id: userCredential.user.uid,
+          }),
+        );
+        navigate('/dashboard');
+      })
+      .catch(console.error);
```
Comment (AdammmVV) [inline src/components/Auth/AuthForm.tsx]: put it in redux-thunk

File: src/components/Auth/AuthForm.tsx [hunk 13]
```diff
@@ -0,0 +1,201 @@
...
+  };
+  const handleLogin = event => {
+    event.preventDefault();
+    const auth = getAuth();
+    const email = form.values.email;
+    const password = form.values.password;
+    signInWithEmailAndPassword(auth, email, password)
```
Comment (AdammmVV) [inline src/components/Auth/AuthForm.tsx]: put it in api file

File: src/components/Auth/AuthForm.tsx [hunk 14]
```diff
@@ -0,0 +1,201 @@
...
+          setUser({
+            login: form.values.login,
+            email,
+            token: userCredential.user.refreshToken,
+            id: userCredential.user.uid,
+          }),
+        );
+        navigate('/dashboard');
+      })
+      .catch(console.error);
```
Comment (AdammmVV) [inline src/components/Auth/AuthForm.tsx]: put it in redux-thunk

File: src/components/Auth/AuthForm.tsx [hunk 15]
```diff
@@ -0,0 +1,201 @@
...
+import { setRole, setUser } from 'redux/slice/userSlice';
+import { useDispatch } from 'react-redux';
+
+interface AuthenticationFormProps extends PaperProps {
+  type: 'login' | 'register';
+}
+export function AuthenticationForm({
+  type: formType,
+  ...props
+}: AuthenticationFormProps) {
```
Comment (AdammmVV) [inline src/components/Auth/AuthForm.tsx]: use Arrow function and `FC<Props>`

File: src/components/Auth/AuthForm.tsx [hunk 16]
```diff
@@ -0,0 +1,201 @@
...
+          {t('Hello')}! {t('Welcome')}!
+        </Text>
+        <Group grow mb="md" mt="md">
+          <GoogleButton radius="xl">Google</GoogleButton>
+          <GitHubButton radius="xl">GitHub</GitHubButton>
+        </Group>
+
+        <Divider label={t('Or continue with email')} labelPosition="center" my="lg" />
+
+        <form onSubmit={type === 'login' ? handleLogin : handleRegister}>
```
Comment (AdammmVV) [inline src/components/Auth/AuthForm.tsx]: use [form.onSubmit](https://mantine.dev/form/use-form/#onreset-and-onsubmit)

File: src/components/Auth/AuthForm.tsx [hunk 17]
```diff
@@ -0,0 +1,201 @@
...
+          <Stack>
+            {type === 'register' && (
+              <TextInput
+                label={t('Name')}
+                placeholder={t('Your name')}
+                value={form.values.name}
+                onChange={event => form.setFieldValue('name', event.currentTarget.value)}
```
Comment (AdammmVV) [inline src/components/Auth/AuthForm.tsx]: use [getInputProps](https://mantine.dev/form/get-input-props/#getinputprops-handler)

File: src/components/shared/AuthButton/GoogleButton.tsx [hunk 18]
```diff
@@ -0,0 +1,102 @@
...
+        fill="#FBBC05"
+        d="M56.281 156.37c-2.756-8.123-4.351-16.827-4.351-25.82 0-8.994 1.595-17.697 4.206-25.82l-.073-1.73L15.26 71.312l-1.335.635C5.077 89.644 0 109.517 0 130.55s5.077 40.905 13.925 58.602l42.356-32.782"
+      />
+      <path
+        fill="#EB4335"
+        d="M130.55 50.479c24.514 0 41.05 10.589 50.479 19.438l36.844-35.974C195.245 12.91 165.798 0 130.55 0 79.49 0 35.393 29.301 13.925 71.947l42.211 32.783c10.59-31.477 39.891-54.251 74.414-54.251"
+      />
+    </svg>
+  );
+}
```
Comment (AdammmVV) [inline src/components/shared/AuthButton/GoogleButton.tsx]: put it in separate file or try [icon](https://mantine.dev/guides/icons/)

File: src/components/shared/AuthButton/GoogleButton.tsx [hunk 19]
```diff
@@ -0,0 +1,102 @@
...
+  props: ButtonProps & React.ComponentPropsWithoutRef<'button'>,
+) {
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const [login, setLogin] = useState('');
+
+  const signInWithGoogle = () => {
+    const auth = getAuth();
+    const provider = new GoogleAuthProvider();
+    signInWithPopup(auth, provider)
```
Comment (AdammmVV) [inline src/components/shared/AuthButton/GoogleButton.tsx]: put it in file api

File: src/components/shared/AuthButton/GoogleButton.tsx [hunk 20]
```diff
@@ -0,0 +1,102 @@
...
+      <path
+        fill="#EB4335"
+        d="M130.55 50.479c24.514 0 41.05 10.589 50.479 19.438l36.844-35.974C195.245 12.91 165.798 0 130.55 0 79.49 0 35.393 29.301 13.925 71.947l42.211 32.783c10.59-31.477 39.891-54.251 74.414-54.251"
+      />
+    </svg>
+  );
+}
+
+export function GoogleButton(
+  props: ButtonProps & React.ComponentPropsWithoutRef<'button'>,
```
Comment (AdammmVV) [inline src/components/shared/AuthButton/GoogleButton.tsx]: use Arrow function and `FC<Props>`

File: src/components/shared/AuthButton/GoogleButton.tsx [hunk 21]
```diff
@@ -0,0 +1,102 @@
...
+                onClick={async () => {
+                  modals.closeAll();
+                  navigate('/dashboard');
+                  const db = getFirestore();
+                  await setDoc(doc(db, 'users', result.user.uid), {
+                    login: login,
+                  });
+                  // Обновляем только логин пользователя в состоянии Redux
+                  dispatch(setLogin(login));
+                }}
```
Comment (AdammmVV) [inline src/components/shared/AuthButton/GoogleButton.tsx]: move the logic for working with the database into a separate file here and throughout the app and all async logic in redux-thunk

File: src/components/shared/AuthButton/GoogleButton.tsx [hunk 22]
```diff
@@ -0,0 +1,102 @@
...
+    signInWithPopup(auth, provider)
+      .then(result => {
+        dispatch(
+          setUser({
+            login: result.user.displayName,
+            email: result.user.email,
+            token: result.user.refreshToken,
+            id: result.user.uid,
+          }),
+        );
```
Comment (AdammmVV) [inline src/components/shared/AuthButton/GoogleButton.tsx]: put it in redux-thunk

File: src/components/shared/AuthButton/GoogleButton.tsx [hunk 23]
```diff
@@ -0,0 +1,102 @@
...
+                    login: login,
+                  });
+                  // Обновляем только логин пользователя в состоянии Redux
+                  dispatch(setLogin(login));
+                }}
+                mt="md"
+              >
+                Submit
+              </Button>
+            </>
```
Comment (AdammmVV) [inline src/components/shared/AuthButton/GoogleButton.tsx]: try put in in sparate component

File: src/components/shared/AuthButton/GoogleButton.tsx [hunk 24]
```diff
@@ -0,0 +1,102 @@
...
+          ),
+        });
+      })
+      .catch(error => {
+        console.error('Ошибка входа в систему:', error);
+      });
```
Comment (AdammmVV) [inline src/components/shared/AuthButton/GoogleButton.tsx]: errors need to be handled. How the user will see it?)

File: src/layout/header/Header.tsx [hunk 25]
```diff
@@ -0,0 +1,57 @@
...
+      <Group justify="space-between" h="100%">
+        <Group justify="center" grow>
+          <Image src={logo} h={30} alt="logo" />
+          <Text fw={700}>ClubApp </Text>
+        </Group>
+
+        <Group justify="center">
+          {isAuthenticated ? (
+            <>
+              <Button onClick={handleLogoutClick}>Log out from</Button>
```
Comment (AdammmVV) [inline src/layout/header/Header.tsx]: add translate here and all app

File: src/hooks/use-auth.ts [hunk 26]
```diff
@@ -0,0 +1,13 @@
+import { useSelector } from 'react-redux';
+
+export const useAuth = () => {
+  const { email, token, id, login } = useSelector(state => state.user);
```
Comment (AdammmVV) [inline src/hooks/use-auth.ts]: do not use [useSelector destructuring](https://stackoverflow.com/questions/59072200/useselector-destructuring-vs-multiple-calls), use useSelector only in the component where it is needed

File: src/components/shared/AuthButton/GoogleButton.tsx [hunk 27]
```diff
@@ -0,0 +1,102 @@
...
+            id: result.user.uid,
+          }),
+        );
+        modals.open({
+          title: 'Введите ваш логин',
+          children: (
+            <>
+              <TextInput
+                label="Ваш логин"
+                placeholder="Ваш логин"
```
Comment (AdammmVV) [inline src/components/shared/AuthButton/GoogleButton.tsx]: use translate

File: src/components/Auth/AuthForm.tsx [hunk 28]
```diff
@@ -0,0 +1,201 @@
...
+    },
+
+    validate: {
+      email: val => (/^\S+@\S+$/.test(val) ? null : 'Invalid email'),
+      password: val =>
+        val.length <= 6 ? 'Password should include at least 6 characters' : null,
+    },
+  });
+
+  const dispatch = useDispatch();
```
Comment (AdammmVV) [inline src/components/Auth/AuthForm.tsx]: use [useAppDispatch](https://redux-toolkit.js.org/usage/usage-with-typescript#getting-the-dispatch-type)

File: src/redux/slice/userSlice.ts [hunk 29]
```diff
@@ -98,7 +167,29 @@ const authSlice = createSlice({
         state.user = action.payload;
       }
     });
-    builder;
+    builder.addCase(updateProfile.fulfilled, (state, action) => {
+      if (action.payload) {
+        state.status = 'succeeded';
+        state.user = action.payload;
+      }
+    });
```
Comment (AdammmVV) [inline src/redux/slice/userSlice.ts]: use the same notation for addCase

```
.addCase(authGoogle.fulfilled, (state, action) => {
         if (action.payload) {
           state.status = 'succeeded';
           state.user = action.payload;
           state.isAuthenticated = true;
         }
       });
```

File: src/components/pages/ProfilePage/Profile.tsx [hunk 30]
```diff
@@ -0,0 +1,50 @@
...
+import { Outlet, useNavigate } from 'react-router-dom';
+import classes from 'components/pages/ProfilePage/Profile.module.css';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+const mockdata = [
+  { label: 'Teams', route: 'teams' },
+  { label: 'Friends', route: 'friends' },
+  { label: 'Statistics', route: 'statistics' },
+];
```
Comment (AdammmVV) [inline src/components/pages/ProfilePage/Profile.tsx]: put it in config

File: src/components/pages/ProfilePage/Profile.tsx [hunk 31]
```diff
@@ -0,0 +1,50 @@
...
+      className={classes.mainLink}
+      data-active={index === active || undefined}
+      onClick={event => {
+        event.preventDefault();
+        setActive(index);
+        navigate(item.route);
+      }}
+    >
+      {t(item.label)}
+    </Anchor>
```
Comment (AdammmVV) [inline src/components/pages/ProfilePage/Profile.tsx]: try use `<Link to="...">...</Link>`

File: src/components/pages/ProfilePage/UploadAvatar.tsx [hunk 32]
```diff
@@ -0,0 +1,86 @@
...
+        <div className={classes.hoverOverlay}>
+          <ActionIcon
+            m="lg"
+            radius="lg"
+            variant="subtle"
+            size="lg"
+            onClick={handleUploadClick}
+          >
+            <IconUpload />
+          </ActionIcon>
```
Comment (AdammmVV) [inline src/components/pages/ProfilePage/UploadAvatar.tsx]: try use [file-button](https://mantine.dev/core/file-button/#usage)

File: src/components/pages/ProfilePage/UploadAvatar.module.css [hunk 33]
```diff
@@ -0,0 +1,21 @@
...
+  position: absolute;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  display: flex;
+  justify-content: center;
+  align-items: flex-end;
+  border-radius: 15%;
+}
```
Comment (AdammmVV) [inline src/components/pages/ProfilePage/UploadAvatar.module.css]: use mantine components and [Style props](https://mantine.dev/styles/style-props/) here and fix in the whole app

File: src/components/pages/ProfilePage/UploadAvatar.tsx [hunk 34]
```diff
@@ -0,0 +1,86 @@
...
+} from 'redux/slice/userSlice';
+import { useAppDispatch } from 'store/store';
+import { useState, useRef, useEffect } from 'react';
+import classes from 'components/pages/ProfilePage/UploadAvatar.module.css';
+import { IconUpload, IconTrashX } from '@tabler/icons-react';
+
+function AvatarUpload() {
+  const dispatch = useAppDispatch();
+  const auth = useAuth();
+  const [isHovered, setIsHovered] = useState(false);
```
Comment (AdammmVV) [inline src/components/pages/ProfilePage/UploadAvatar.tsx]: try use [use-hover](https://mantine.dev/hooks/use-hover/#usage)

File: src/components/pages/ProfilePage/UploadAvatar.tsx [hunk 35]
```diff
@@ -0,0 +1,86 @@
...
+
+  const handleFileChange = async (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (event.target.files && event.target.files.length > 0) {
+      const file = event.target.files[0];
+      const actionResult = await dispatch(uploadAvatar(file));
+      if (uploadAvatar.fulfilled.match(actionResult)) {
+        const avatarUrl = actionResult.payload;
+        dispatch(saveUserAvatar({ userId: auth.id, avatarUrl }));
+      }
+    }
```
Comment (AdammmVV) [inline src/components/pages/ProfilePage/UploadAvatar.tsx]: why not do it all in Thunk?

File: src/components/pages/ProfilePage/UploadAvatar.tsx [hunk 36]
```diff
@@ -0,0 +1,86 @@
...
+
+  return (
+    <Paper
+      onMouseEnter={() => setIsHovered(true)}
+      onMouseLeave={() => setIsHovered(false)}
+      className={classes.avatarContainer}
+    >
+      <Avatar src={auth.avatarUrl} size={200} radius="xl" />
+      {isHovered && (
+        <div className={classes.hoverOverlay}>
```
Comment (AdammmVV) [inline src/components/pages/ProfilePage/UploadAvatar.tsx]: use mantine component

File: src/components/pages/ProfilePage/UploadAvatar.tsx [hunk 37]
```diff
@@ -0,0 +1,86 @@
...
+            <IconTrashX />
+          </ActionIcon>
+        </div>
+      )}
+      <input
+        type="file"
+        ref={fileInputRef}
+        onChange={handleFileChange}
+        className={classes.hiddenInput}
+      />
```
Comment (AdammmVV) [inline src/components/pages/ProfilePage/UploadAvatar.tsx]: try to fix it using the [file-button](https://mantine.dev/core/file-button/#usage)

File: src/components/pages/ProfilePage/UserInfo.tsx [hunk 38]
```diff
@@ -0,0 +1,90 @@
...
+            p="md"
+            placeholder={t('userinfo.Your Last Name')}
+            data-autofocus
+            onChange={event => (lastName = event.currentTarget.value)}
+          />
+
+          <Button fullWidth onClick={handleProfileUpdate} mt="md">
+            {t('userinfo.Submit')}
+          </Button>
+        </>
```
Comment (AdammmVV) [inline src/components/pages/ProfilePage/UserInfo.tsx:46]: put this in a special component

File: src/components/pages/ProfilePage/UserInfo.tsx [hunk 39]
```diff
@@ -0,0 +1,90 @@
...
+      ),
+    });
+  };
+
+  return (
+    <>
+      <div className={classes.wrapper}>
```
Comment (AdammmVV) [inline src/components/pages/ProfilePage/UserInfo.tsx]: same mantine component

File: src/components/pages/ProfilePage/UserInfo.tsx [hunk 40]
```diff
@@ -0,0 +1,90 @@
...
+        <Paper className={classes.paper}>
+          <AvatarUpload />
+        </Paper>
+
+        <Group p="md" wrap="nowrap">
+          <div className={classes.item}>
```
Comment (AdammmVV) [inline src/components/pages/ProfilePage/UserInfo.tsx]: and here

File: src/components/shared/LanguageSwitcher/LanguageSwitcher.tsx [hunk 41]
```diff
@@ -1,34 +1,59 @@
...
+  const [opened, setOpened] = useState(false);
+  const [selected, setSelected] = useState(initialLanguage);
+
+  const items = data.map(item => (
+    <Menu.Item
+      leftSection={<Image src={item.image} width={35} height={25} />}
+      onClick={() => {
+        setSelected(item);
+        changeLanguage(item.code as langSwitcher);
+      }}
```
Comment (AdammmVV) [inline src/components/shared/LanguageSwitcher/LanguageSwitcher.tsx]: put it in function

File: src/utils/constsRoutes.ts [hunk 42]
```diff
@@ -4,3 +4,7 @@ export const DASHBOARD_ROUTE = '/dashboard';
 export const HOME_ROUTE = '/';
 export const ADMIN_ROUTE = '/admin';
 export const NOT_FOUND_ROUTE = '/404';
+export const PROFILE_ROUTE = '/profile';
+export const PROFILE_FRIENDS_ROUTE = 'friends';
+export const PROFILE_STATISTICS_ROUTE = 'statistics';
+export const PROFILE_TEAMS_ROUTE = 'teams';
```
Comment (AdammmVV) [inline src/utils/constsRoutes.ts]: try create Enum `pths` for this

File: src/components/shared/LanguageSwitcher/LanguageSwitcher.tsx [hunk 43]
```diff
@@ -0,0 +1,63 @@
...
+  };
+
+  const items = data.map(item => (
+    <Menu.Item
+      leftSection={<Image src={item.image} width={35} height={25} />}
+      onClick={setLang(item)}
+      key={item.label}
+    >
+      {item.label}
+    </Menu.Item>
```
Comment (AdammmVV) [inline src/components/shared/LanguageSwitcher/LanguageSwitcher.tsx:35]: put this in markup

File: src/components/shared/LanguageSwitcher/LanguageSwitcher.tsx [hunk 44]
```diff
@@ -0,0 +1,63 @@
...
+import classes from 'components/shared/LanguageSwitcher/LanguageSwitcher.module.css';
+import { useTranslation } from 'react-i18next';
+import { langSwitcher } from 'Enum/Enum';
+
+const data = [
+  { label: 'Русский', image: ru, code: 'ru' },
+  { label: 'English', image: en, code: 'en' },
+];
+
+function LanguageSwitcher() {
```
Comment (AdammmVV) [inline src/components/shared/LanguageSwitcher/LanguageSwitcher.tsx:14]: use arrow function and throughout app
```suggestion
export const LanguageSwitcher: FC = () => {
```

File: src/components/shared/LanguageSwitcher/LanguageSwitcher.tsx [hunk 45]
```diff
@@ -0,0 +1,63 @@
...
+      key={item.label}
+    >
+      {item.label}
+    </Menu.Item>
+  ));
+
+  function changeLanguage(language: langSwitcher) {
+    i18n.changeLanguage(language);
+    localStorage.setItem('language', language);
+  }
```
Comment (AdammmVV) [inline src/components/shared/LanguageSwitcher/LanguageSwitcher.tsx:41]: use arrow function

File: src/components/shared/LanguageSwitcher/LanguageSwitcher.tsx [hunk 46]
```diff
@@ -0,0 +1,63 @@
...
+import en from 'components/shared/LanguageSwitcher/Image/english.png';
+import ru from 'components/shared/LanguageSwitcher/Image/russian.png';
+import classes from 'components/shared/LanguageSwitcher/LanguageSwitcher.module.css';
+import { useTranslation } from 'react-i18next';
+import { langSwitcher } from 'Enum/Enum';
+
+const data = [
+  { label: 'Русский', image: ru, code: 'ru' },
+  { label: 'English', image: en, code: 'en' },
+];
```
Comment (AdammmVV) [inline src/components/shared/LanguageSwitcher/LanguageSwitcher.tsx:12]: put it separatu file config

File: src/layout/header/Header.tsx [hunk 47]
```diff
@@ -0,0 +1,84 @@
...
+  const handleLoginClick = () => {
+    navigate(RoutePaths.LOGIN_ROUTE);
+  };
+
+  const handleSignupClick = () => {
+    navigate(RoutePaths.REGISTRATION_ROUTE);
+  };
+
+  const handleHomeClick = () => {
+    navigate('/');
```
Comment (AdammmVV) [inline src/layout/header/Header.tsx:30]: use Enum for paths

File: src/layout/navbar/NavBar.tsx [hunk 48]
```diff
@@ -0,0 +1,84 @@
...
+  onClick?(): void;
+}
+
+function NavbarLink({ icon: Icon, label, active, onClick }: NavbarLinkProps) {
+  return (
+    <Tooltip label={label} position="right" transitionProps={{ duration: 0 }}>
+      <UnstyledButton
+        onClick={onClick}
+        className={`${classes.link} ${active ? classes.active : ''}`}
+        data-active={active || undefined}
```
Comment (AdammmVV) [inline src/layout/navbar/NavBar.tsx:23]: try to do it like this

```suggestion
        data-active={active}
```

File: src/layout/navbar/NavBar.tsx [hunk 49]
```diff
@@ -0,0 +1,84 @@
...
+          {links}
+        </Stack>
+      </div>
+
+      <Stack justify="center" gap={0}>
+        <NavbarLink icon={IconLogout} label="Logout" onClick={handleLogout} />
+      </Stack>
+    </nav>
+  );
+}
```
Comment (AdammmVV) [inline src/layout/navbar/NavBar.tsx:84]: put it in separete component

File: src/layout/navbar/NavBar.tsx [hunk 50]
```diff
@@ -0,0 +1,84 @@
...
+        <Icon style={{ width: rem(20), height: rem(20) }} stroke={1.5} />
+      </UnstyledButton>
+    </Tooltip>
+  );
+}
+
+const mockdata = [
+  { icon: IconGauge, label: 'Dashboard', route: RoutePaths.DASHBOARD_ROUTE },
+  { icon: IconFingerprint, label: 'Security', route: RoutePaths.ADMIN_ROUTE },
+];
```
Comment (AdammmVV) [inline src/layout/navbar/NavBar.tsx:34]: put it in file config

File: src/hooks/useAuth.ts [hunk 51]
```diff
@@ -0,0 +1,6 @@
+import { useSelector } from 'react-redux';
+import { RootState } from 'store/store';
+
+export const useAuth = () => {
+  return useSelector((state: RootState) => state.user);
+};
```
Comment (AdammmVV) [inline src/hooks/useAuth.ts:6]: put it in selectors `(state: RootState) => state.user`
use useAppSelector don't create unnecessary hooks

File: src/layout/navbar/NavBar.tsx [hunk 52]
```diff
@@ -0,0 +1,84 @@
+import { useEffect } from 'react';
+import { Tooltip, UnstyledButton, Stack, rem } from '@mantine/core';
+import { IconHome2, IconGauge, IconFingerprint, IconLogout } from '@tabler/icons-react';
+import classes from './NavBar.module.css';
+import { RoutePaths } from 'Enum/Enum';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { useAuth } from 'hooks/useAuth';
+import { persistor, useAppDispatch } from 'store/store';
+import { logout } from 'redux/slice/userSlice';
+interface NavbarLinkProps {
```
Comment (AdammmVV) [inline src/layout/navbar/NavBar.tsx:10]: add emty line and use type Props for all component:
```suggestion
type Props = {
```

File: src/redux/slice/userSlice.ts [hunk 53]
```diff
@@ -0,0 +1,195 @@
...
+    try {
+      const user = await signInWithGoogle(login);
+      return user;
+    } catch (error: unknown) {
+      return thunkAPI.rejectWithValue(error);
+    }
+  },
+);
+
+export const updateProfile = createAsyncThunk(
```
Comment (AdammmVV) [inline src/redux/slice/userSlice.ts:62]: add types for createAsyncThunk

File: src/redux/slice/userSlice.ts [hunk 54]
```diff
@@ -0,0 +1,195 @@
...
+  ) => {
+    try {
+      const updatedUser = await updateUserProfile(user, firstName, lastName);
+      return updatedUser;
+    } catch (error: unknown) {
+      return thunkAPI.rejectWithValue(error);
+    }
+  },
+);
+export const updateLogin = createAsyncThunk(
```
Comment (AdammmVV) [inline src/redux/slice/userSlice.ts:76]: add emty line add types for createAsyncThunk

File: src/redux/slice/userSlice.ts [hunk 55]
```diff
@@ -0,0 +1,195 @@
...
+    try {
+      const updatedUser = await updateUserLogin(user, newLogin);
+      return updatedUser;
+    } catch (error: unknown) {
+      return thunkAPI.rejectWithValue(error);
+    }
+  },
+);
+
+export const uploadAvatar = createAsyncThunk(
```
Comment (AdammmVV) [inline src/redux/slice/userSlice.ts:88]: and here

File: src/redux/slice/userSlice.ts [hunk 56]
```diff
@@ -0,0 +1,195 @@
...
+    try {
+      const url = await uploadFile(state.user, file, state.user.id);
+      return url;
+    } catch (error: unknown) {
+      return thunkAPI.rejectWithValue(error);
+    }
+  },
+);
+
+export const deleteUserAvatar = createAsyncThunk(
```
Comment (AdammmVV) [inline src/redux/slice/userSlice.ts:101]: and here

File: src/redux/slice/userSlice.ts [hunk 57]
```diff
@@ -0,0 +1,195 @@
...
+  'user/deleteAvatar',
+  async (avatarUrl: string) => {
+    await deleteAvatar(avatarUrl);
+    return avatarUrl;
+  },
+);
+
+export const logout = createAsyncThunk('user/logout', async () => {
```
Comment (AdammmVV) [inline src/redux/slice/userSlice.ts:109]: and here

File: src/redux/slice/userSlice.ts [hunk 58]
```diff
@@ -0,0 +1,195 @@
...
+        state.user = action.payload;
+        state.isAuthenticated = true;
+      }
+    });
+    builder.addCase(updateProfile.fulfilled, (state, action) => {
+      if (action.payload) {
+        state.status = 'succeeded';
+        state.user = action.payload;
+        state.isAuthenticated = true;
+      }
```
Comment (AdammmVV) [inline src/redux/slice/userSlice.ts:154]: try use addMacher for recurring actions and use:
```suggestion
      .addCase(authGoogle.fulfilled, (state, action) => {
        if (action.payload) {
          state.status = 'succeeded';
          state.user = action.payload;
          state.isAuthenticated = true;
        }
      })
    .addCase(updateLogin.fulfilled, (state, action) => {
      if (action.payload) {
        state.status = 'succeeded';
        state.user = action.payload;
        state.isAuthenticated = true;
      }
    })
    .addCase(updateProfile.fulfilled, (state, action) => {
      if (action.payload) {
        state.status = 'succeeded';
        state.user = action.payload;
        state.isAuthenticated = true;
      }
```

**Comment by alexspel** [comment]
+

**Pull request merged**